src/transformers/modeling_flax_pytorch_utils.py

```diff
@@ -118,13 +128,23 @@ def convert_pytorch_state_dict_to_flax(pt_state_dict, flax_model):
     pt_state_dict = {k: v.numpy() for k, v in pt_state_dict.items()}
 
     model_prefix = flax_model.base_model_prefix
-    random_flax_state_dict = flatten_dict(flax_model.params)
+
+    # use params dict if the model contains batch norm layers and add batch_stats keys,values to dict
+    if "params" in flax_model.params and "batch_stats" in flax_model.params:
```

Comment:
Could we not always set:
```python
flax_model_params = flax_model.params["params"]
random_flax_state_dict = flatten_dict(flax_model_params)
```
And then check if `"batch_stats"` is in the flax params dict:
```python
if "batch_stats" in flax_model.params:
    flax_batch_stats = flatten_dict(flax_model.params["batch_stats"])
    random_flax_state_dict.update(flax_batch_stats)
```

Just to simplify the `if/else` logic a bit. If I understand correctly then we'll always have `"params"` in the params dict, and only `"batch_stats"` if using batch norm (as per https://flax.readthedocs.io/en/latest/guides/batch_norm.html#the-batch-stats-collection).

Reply:
I added params checking in `if/else` just to extra make sure we have batch norm layers, but I agree it can be simplified, made the changes in the most recent commit.